terminusdb_client/woqlclient/woqlClient.py

```diff
@@ -368,7 +371,7 @@ def uid(self, ignore_jwt=True):
         """
         return self.conConfig.user(ignore_jwt)
 
-    def resource(self, ttype, val=None):
+    def resource(self, ttype: str, val: Optional[str] = None) -> str:
```

Comment:
Would it be better to make `ttype` an `Enum` here, given that it can only take a limited number of values? It would help prevent errors due to typos and such directly when writing code, instead of finding them at execution.